snc-library.sh

```diff
@@ -10,6 +10,29 @@ function preflight_failure() {
         fi
 }
 
+function download_oc_binary() {
+        mkdir -p openshift-clients/linux
+
+        if [ "${ARCH}" != "ppc64le" ]; then
+                mkdir openshift-clients/mac openshift-clients/windows
+        fi
+
```

Comment:
I think I'd move these `mkdir` where they are needed, even if this means duplicating them:
```
 if [[ ${OKD_VERSION} != "none" ]]; then
                mkdir -p openshift-clients/linux
                curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux-${OPENSHIFT_RELEASE_VERSION}.tar.gz" | tar -zx -C openshift-clients/linux oc
                if [ "${ARCH}" != "ppc64le" ]; then
                        curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-mac-${OPENSHIFT_RELEASE_VERSION}.tar.gz" | tar -zx -C openshift-clients/mac oc
                        mkdir -p openshift-clients/windows
                        curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-windows-${OPENSHIFT_RELEASE_VERSION}.zip" > openshift-clients/windows/oc.zip
                fi
        else
                curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux.tar.gz" | tar -zx -C openshift-clients/linux oc
                if [ "${ARCH}" != "ppc64le" ]; then
                        curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-mac.tar.gz" | tar -zx -C openshift-clients/mac oc
                        mkdir -p openshift-clients/windows
                        curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-windows.zip" > openshift-clients/windows/oc.zip
                        ${UNZIP} -o -d openshift-clients/windows/ openshift-clients/windows/oc.zip
                fi
        fi
```
The `mkdir openshift-clients/mac` was removed?